hphp/runtime/vm/jit/irlower-call.cpp

```diff
@@ -477,16 +477,24 @@ void cgCheckRefs(IRLS& env, const IRInstruction* inst)  {
       auto const bits = v.makeReg();
       v << load{bitsPtr[bitsOff], bits};
 
-      auto const truncBits = v.makeReg();
+      /*
+       * When the mask is gt/eq to the value,
+       * then no truncation of bits is needed.
+       */
+      assertx(mask64 >= vals64);
+
+      /* maskedBits = and(bits, mask64) */
       auto const maskedBits = v.makeReg();
 
-      if (mask64 <= 0xff && vals64 <= 0xff) {
-        v << movtqb{bits, truncBits};
-        v << andbi{(int8_t)mask64, truncBits, maskedBits, v.makeReg()};
+      if (mask64 < 0xff && vals64 <= 0xff) {
+        v << andbi{(int8_t)mask64, bits, maskedBits, v.makeReg()};
         v << cmpbi{(int8_t)vals64, maskedBits, sf};
+      } else if (mask64 == 0xff && vals64 <= 0xff) {
+        v << cmpbi{(int8_t)vals64, bits, sf};
       } else if (mask64 <= 0xffffffff && vals64 <= 0xffffffff) {
```

Comment:
Is this supposed to be changed to `} else if (mask64 < 0xffffffff && vals64 <= 0xffffffff) {` now, similar to the first change?